#!/usr/bin/env python
PACKAGE="ray_ground_filter"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("sensor_height", double_t, 0, "sensor_height", 1.7, 0.3, 2.0)
gen.add("radial_divider_angle", double_t, 0, "radial_divider_angle", 0.18, 0.1, 0.5)
gen.add("concentric_divider_distance", double_t, 0, "concentric_divider_distance", 0.02, 0.01, 0.12)
gen.add("local_max_slope", double_t, 0, "local_max_slope", 8.0, 1.0, 10.0)
gen.add("general_max_slope", double_t, 0, "general_max_slope", 7.0, 1.0, 10.0)
gen.add("min_height_threshold", double_t, 0, "min_height_threshold", 0.05, 0.02, 0.2)
gen.add("reclass_distance_threshold", double_t, 0, "reclass_distance_threshold", 0.2, 0.1, 1.0)
gen.add("clip_height", double_t, 0, "clip_height", 0.2, -0.5, 1.0)
gen.add("min_distance", double_t, 0, "min_distance", 0.5, 0.2, 2.0)
gen.add("minX", double_t, 0, "minX", -0.5, -2.0, 0)
gen.add("maxX", double_t, 0, "maxX", 20, 10, 80)
gen.add("minY", double_t, 0, "minY", -10, -20, 0)
gen.add("maxY", double_t, 0, "maxY", 10, 10, 80)
gen.add("remove_min_x", double_t, 0, "remove_min_x", -2.0, -3.0, 0)
gen.add("remove_min_y", double_t, 0, "remove_min_y", -0.5, -2.0, 0)
gen.add("remove_max_x", double_t, 0, "remove_max_x", 0, 0, 2.0)
gen.add("remove_max_y", double_t, 0, "remove_max_y", 0.5, 0, 2.0)

exit(gen.generate(PACKAGE, "ray_ground_filter_node", "RayFilter"))